_studio/mfx_lib/encode_hw/hevc/agnostic/base/hevcehw_base_legacy.cpp

```diff
@@ -2585,7 +2585,7 @@ void Legacy::SetVPS(
     general.progressive_source_flag         = !!(par.mfx.FrameInfo.PicStruct & MFX_PICSTRUCT_PROGRESSIVE);
     general.interlaced_source_flag          = !(par.mfx.FrameInfo.PicStruct & MFX_PICSTRUCT_PROGRESSIVE);
     general.non_packed_constraint_flag      = 0;
-    general.frame_only_constraint_flag      = 0;
+    general.frame_only_constraint_flag      = !(par.mfx.FrameInfo.PicStruct & MFX_PICSTRUCT_FIELD_SINGLE);
```

Comment:
I'd always set it to 1 here and force to 0 there:
https://github.com/Intel-Media-SDK/MediaSDK/blob/68c7cb74c47cd2c15b2719ff98a0a2c491603c77/_studio/mfx_lib/encode_hw/hevc/agnostic/base/hevcehw_base_interlace.cpp#L237-L257

Reply:
Good point. Let me fix.

Reply:
@eistarov , I'm a bit confused. In defaults.GetSPS.Push we modify sps, can't touch vps. And per https://github.com/Intel-Media-SDK/MediaSDK/blob/908cc8bce852d9dac4c7d53132180a6d59a1cc71/_studio/mfx_lib/encode_hw/hevc/agnostic/base/hevcehw_base_legacy_defaults.cpp#L1859
it seems the in current design sps.general always inherits values from vps.general. So if we move the fix to hevcehw_base_interlace.cpp, won't we "break" the original design? I mean VPS.general won't be equal to SPS.general anymore.

Reply:
So, the "right" solution would be to move logic from Legacy::SetVPS to Defaults::GetVPS (similar to GetSPS/GetPPS) and push extra VPS logic in interlace feature: defaults.GetVPS.Push([]( ...

Reply:
Okay

Reply:
@eistarov , updated. Could you take a look?